src/Session.cc

```diff
@@ -575,9 +585,16 @@ const AddressSpace::Mapping Session::recreate_shared_mmap(
     new_map = remote.task()->vm()->mapping_of(new_addr);
     if (preserved_data) {
       memcpy(new_map.local_addr, preserved_data, size);
-      munmap(preserved_data, size);
     }
```

Comment:
This here is very problematic as it could unmap the local memory for the wrong task. We want to unmap for `remote.task()`Mapping's `local_addr` and _not_ for Mapping `m`'s `local_addr`